src/plugins_3rdparty/primer3/src/Primer3Task.cpp

```diff
@@ -663,12 +663,19 @@ void Primer3Task::selectPairsSpanningIntron(p3retval* primers, int toReturn) {
 
 // Primer3SWTask
 
-Primer3SWTask::Primer3SWTask(Primer3TaskSettings* _settings)
+Primer3SWTask::Primer3SWTask(Primer3TaskSettings* _settings, bool _ownsSettings)
     : Task("Pick primers SW task", TaskFlags_NR_FOSCOE | TaskFlag_CollectChildrenWarnings),
-      settings(_settings) {
+      settings(_settings),
+      ownsSettings(_ownsSettings) {
     median = settings->getSequenceSize() / 2;
 }
 
+Primer3SWTask::~Primer3SWTask() {
+    if (ownsSettings) {
+        delete settings;
```

Comment:
Can you just pass the settings by value instead of these complex memory management protocol?

Reply:
What will happen if you pass `Primer3TaskSettings` by value? So it will always be copied.
Will it break anything?

Reply:
Dima purposefully changed pass by value to pass by pointer. I don't think it's correct to change this behavior back... So, I only have the QD that uses `Primer3SWTask`, but this task now takes a pointer to the settings, not the settings themselves. And I needed to change the behavior according to the new logic. This is how I see my task.

Why is a pointer used? I think Primer3's settings constructor became [non-trivial](https://github.com/ugeneunipro/ugene/blob/d407dc1a0cce97d9a3d597fab117eac8b5ec1cbd/src/plugins_3rdparty/primer3/src/Primer3TaskSettings.cpp#L48-L49), so copying became [forbidden](https://github.com/ugeneunipro/ugene/blob/d407dc1a0cce97d9a3d597fab117eac8b5ec1cbd/src/plugins_3rdparty/primer3/src/Primer3TaskSettings.h#L55-L58). It was decided to work around this with raw pointers.

Reply:
ok